src/averageAges.js

```diff
@@ -15,11 +15,14 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const males = people
+    .filter(person => person.sex.toLowerCase() === 'm'
+      && (!century || Math.ceil(person.died / 100) === century));
+
+  return males.reduce(
+    (acc, currentPerson) => {
+      return acc + currentPerson.died - currentPerson.born;
+    }, 0) / males.length;
```

Comment:
There's a principle in programming called DRY - don't repeat yourself. If you do something twice or more than twice you should move it to a function.

Reply:
Can you give some advice. Is it appropriate to use expression as a parameter of my custom function avarageAge()?
<img width="546" alt="image" src="https://user-images.githubusercontent.com/88326325/228930936-1f8a8cc2-22fb-4c0d-83f9-76e18cbfc134.png">